packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -1156,8 +1162,43 @@ export class CdkToolkit {
       defaultBehavior: DefaultSelection.OnlySingle,
     });
 
-    // No validation
+    const selectorWithoutPatterns: StackSelector = {
+      ...selector,
+      allTopLevel: true,
+      patterns: [],
+    };
+    const stacksWithoutPatterns = await assembly.selectStacks(selectorWithoutPatterns, {
+      extend: exclusively ? ExtendedStackSelection.None : ExtendedStackSelection.Downstream,
+      defaultBehavior: DefaultSelection.OnlySingle,
+    });
+
+    const patterns = selector.patterns.map(pattern => {
+      const notExist = !stacks.stackArtifacts.find(stack =>
+        minimatch(stack.hierarchicalId, pattern) || (stack.id === pattern && semver.major(versionNumber()) < 2),
+      );
+
+      const closelyMatched = notExist ? stacksWithoutPatterns.stackArtifacts.map(stack => {
+        if (minimatch(stack.hierarchicalId.toLowerCase(), pattern.toLowerCase())) {
+          return stack.hierarchicalId;
+        }
+        if (stack.id.toLowerCase() === pattern.toLowerCase() && semver.major(versionNumber()) < 2) {
+          return stack.id;
+        }
+        return;
+      }).filter((stack): stack is string => stack !== undefined) : [];
+      return {
+        pattern,
+        notExist,
+        closelyMatched,
+      };
+    });
 
+    patterns.forEach(pattern => {
+      if (pattern.notExist) {
+        const closelyMatched = pattern.closelyMatched.length > 0 ? ` Do you mean ${pattern.closelyMatched.join(', ')}?` : '';
+        warning(`${pattern.pattern} does not exist.${closelyMatched}`);
+      }
+    });
```

Comment:
So this whole section is about finding similarly named stacks (currently only ignoring case I think). We should move this into a separate method `suggestStacks()` or something like that, that can be used elsewhere in future.

The logic to detect that no stacks where found and to suggest typos should be inside the action (here: `destroy`) not in `selectStacksForDestroy`.

Reply:
Changed: https://github.com/aws/aws-cdk/pull/32636/commits/08a9fdd80019deae938a46fc57c73f20fa4553a8